<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Welcome to PyHPO’s documentation! &mdash; PyHPO 3.1.4 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/sphinx_highlight.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="HPOTerm" href="hpoterm.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="#" class="icon icon-home">
            PyHPO
          </a>
              <div class="version">
                3.1.4
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">API documentation:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="hpoterm.html">HPOTerm</a></li>
<li class="toctree-l1"><a class="reference internal" href="ontology.html">Ontology</a></li>
<li class="toctree-l1"><a class="reference internal" href="annotations.html">Annotations</a></li>
<li class="toctree-l1"><a class="reference internal" href="set.html">HPOSet</a></li>
<li class="toctree-l1"><a class="reference internal" href="stats.html">Statistics</a></li>
<li class="toctree-l1"><a class="reference internal" href="similarity.html">Similarity</a></li>
<li class="toctree-l1"><a class="reference internal" href="matrix.html">Matrix</a></li>
<li class="toctree-l1"><a class="reference internal" href="data.html">Handling master data</a></li>
<li class="toctree-l1"><a class="reference internal" href="parser.html">Parser</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">PyHPO</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="#" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Welcome to PyHPO’s documentation!</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/index.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="welcome-to-pyhpo-s-documentation">
<h1>Welcome to PyHPO’s documentation!<a class="headerlink" href="#welcome-to-pyhpo-s-documentation" title="Permalink to this heading"></a></h1>
<div class="toctree-wrapper compound">
<p class="caption" role="heading"><span class="caption-text">API documentation:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="hpoterm.html">HPOTerm</a></li>
<li class="toctree-l1"><a class="reference internal" href="ontology.html">Ontology</a></li>
<li class="toctree-l1"><a class="reference internal" href="annotations.html">Annotations</a></li>
<li class="toctree-l1"><a class="reference internal" href="set.html">HPOSet</a></li>
<li class="toctree-l1"><a class="reference internal" href="stats.html">Statistics</a></li>
<li class="toctree-l1"><a class="reference internal" href="similarity.html">Similarity</a></li>
<li class="toctree-l1"><a class="reference internal" href="matrix.html">Matrix</a></li>
<li class="toctree-l1"><a class="reference internal" href="data.html">Handling master data</a></li>
<li class="toctree-l1"><a class="reference internal" href="parser.html">Parser</a></li>
</ul>
</div>
<section id="introduction">
<h2>Introduction:<a class="headerlink" href="#introduction" title="Permalink to this heading"></a></h2>
</section>
<section id="pyhpo">
<h2>PyHPO<a class="headerlink" href="#pyhpo" title="Permalink to this heading"></a></h2>
<p>A Python library to work with, analyze, filter and inspect the <a class="reference external" href="https://hpo.jax.org/">Human Phenotype Ontology</a></p>
<p>Visit the <a class="reference external" href="https://centogene.github.io/pyhpo/">PyHPO Documentation</a> for a more detailed overview of all the functionality.</p>
<section id="main-features">
<h3>Main features<a class="headerlink" href="#main-features" title="Permalink to this heading"></a></h3>
<ul class="simple">
<li><p>Identify patient cohorts based on clinical features</p></li>
<li><p>Cluster patients or other clinical information for GWAS</p></li>
<li><p>Phenotype to Genotype studies</p></li>
<li><p>HPO similarity analysis</p></li>
<li><p>Graph based analysis of phenotypes, genes and diseases</p></li>
</ul>
<p><strong>PyHPO</strong> allows working on individual terms <code class="docutils literal notranslate"><span class="pre">HPOTerm</span></code>, a set of terms <code class="docutils literal notranslate"><span class="pre">HPOSet</span></code> and the full <code class="docutils literal notranslate"><span class="pre">Ontology</span></code>.</p>
<p>The library is helpful for discovery of novel gene-disease associations and GWAS data analysis studies. At the same time, it can be used for oragnize clinical information of patients in research or diagnostic settings.</p>
<p>Internally the ontology is represented as a branched linked list, every term contains pointers to its parent and child terms. This allows fast tree traversal functionality.</p>
<p>It provides an interface to create <code class="docutils literal notranslate"><span class="pre">Pandas</span> <span class="pre">Dataframe</span></code> from its data, allowing integration in already existing data anlysis tools.</p>
<section id="examples">
<h4>Examples<a class="headerlink" href="#examples" title="Permalink to this heading"></a></h4>
<section id="how-similar-are-the-phenotypes-of-two-patients">
<h5>How similar are the phenotypes of two patients<a class="headerlink" href="#how-similar-are-the-phenotypes-of-two-patients" title="Permalink to this heading"></a></h5>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pyhpo</span> <span class="kn">import</span> <span class="n">Ontology</span>

<span class="c1"># initilize the Ontology ()</span>
<span class="n">_</span> <span class="o">=</span> <span class="n">Ontology</span><span class="p">()</span>

<span class="c1"># Declare the clinical information of the patients</span>
<span class="n">patient_1</span> <span class="o">=</span> <span class="n">HPOSet</span><span class="o">.</span><span class="n">from_queries</span><span class="p">([</span>
    <span class="s1">&#39;HP:0002943&#39;</span><span class="p">,</span>
    <span class="s1">&#39;HP:0008458&#39;</span><span class="p">,</span>
    <span class="s1">&#39;HP:0100884&#39;</span><span class="p">,</span>
    <span class="s1">&#39;HP:0002944&#39;</span><span class="p">,</span>
    <span class="s1">&#39;HP:0002751&#39;</span>
<span class="p">])</span>

<span class="n">patient_2</span> <span class="o">=</span> <span class="n">HPOSet</span><span class="o">.</span><span class="n">from_queries</span><span class="p">([</span>
    <span class="s1">&#39;HP:0002650&#39;</span><span class="p">,</span>
    <span class="s1">&#39;HP:0010674&#39;</span><span class="p">,</span>
    <span class="s1">&#39;HP:0000925&#39;</span><span class="p">,</span>
    <span class="s1">&#39;HP:0009121&#39;</span>
<span class="p">])</span>

<span class="c1"># and compare their similarity</span>
<span class="n">patient_1</span><span class="o">.</span><span class="n">similarity</span><span class="p">(</span><span class="n">patient_2</span><span class="p">)</span>
<span class="c1">#&gt; 0.7594183905785477</span>
</pre></div>
</div>
</section>
<section id="how-close-are-two-hpo-terms">
<h5>How close are two HPO terms<a class="headerlink" href="#how-close-are-two-hpo-terms" title="Permalink to this heading"></a></h5>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pyhpo</span> <span class="kn">import</span> <span class="n">Ontology</span>

<span class="c1"># initilize the Ontology ()</span>
<span class="n">_</span> <span class="o">=</span> <span class="n">Ontology</span><span class="p">()</span>

<span class="n">term_1</span> <span class="o">=</span> <span class="n">Ontology</span><span class="o">.</span><span class="n">get_hpo_object</span><span class="p">(</span><span class="s1">&#39;Scoliosis&#39;</span><span class="p">)</span>
<span class="n">term_2</span> <span class="o">=</span> <span class="n">Ontology</span><span class="o">.</span><span class="n">get_hpo_object</span><span class="p">(</span><span class="s1">&#39;Abnormal axial skeleton morphology&#39;</span><span class="p">)</span>

<span class="n">path</span> <span class="o">=</span> <span class="n">term_1</span><span class="o">.</span><span class="n">path_to_other</span><span class="p">(</span><span class="n">term_2</span><span class="p">)</span>
<span class="k">for</span> <span class="n">t</span> <span class="ow">in</span> <span class="n">path</span><span class="p">[</span><span class="mi">1</span><span class="p">]:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">t</span><span class="p">)</span>

<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">HP:0002650 | Scoliosis</span>
<span class="sd">HP:0010674 | Abnormality of the curvature of the vertebral column</span>
<span class="sd">HP:0000925 | Abnormality of the vertebral column</span>
<span class="sd">HP:0009121 | Abnormal axial skeleton morphology</span>
<span class="sd">&quot;&quot;&quot;</span>
</pre></div>
</div>
</section>
</section>
</section>
<section id="getting-started">
<h3>Getting started<a class="headerlink" href="#getting-started" title="Permalink to this heading"></a></h3>
<p>The easiest way to install <strong>PyHPO</strong> is via pip</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pip<span class="w"> </span>install<span class="w"> </span>pyhpo
</pre></div>
</div>
<p>or, you can additionally install optional packages for extra functionality</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># Include pandas during install</span>
pip<span class="w"> </span>install<span class="w"> </span>pyhpo<span class="o">[</span>pandas<span class="o">]</span>

<span class="c1"># Include scipy</span>
pip<span class="w"> </span>install<span class="w"> </span>pyhpo<span class="o">[</span>scipy<span class="o">]</span>

<span class="c1"># Include all dependencies</span>
pip<span class="w"> </span>install<span class="w"> </span>pyhpo<span class="o">[</span>all<span class="o">]</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Some features of PyHPO require <code class="docutils literal notranslate"><span class="pre">pandas</span></code> and <code class="docutils literal notranslate"><span class="pre">scipy</span></code>. The standard installation via pip will not include pandas or scipy and PyHPO will work just fine. (You will get a warning on the initial import though).</p>
<p>Without installing <code class="docutils literal notranslate"><span class="pre">pandas</span></code>, you won’t be able to export the Ontology as a <code class="docutils literal notranslate"><span class="pre">Dataframe</span></code>, everything else will work fine.</p>
<p>Without installing <code class="docutils literal notranslate"><span class="pre">scipy</span></code>, you won’t be able to use the <code class="docutils literal notranslate"><span class="pre">stats</span></code> module, especially the enrichment calculations.</p>
</div>
</section>
<section id="usage-example">
<h3>Usage example<a class="headerlink" href="#usage-example" title="Permalink to this heading"></a></h3>
<section id="hpoterm">
<h4>HPOTerm<a class="headerlink" href="#hpoterm" title="Permalink to this heading"></a></h4>
<p>An <code class="docutils literal notranslate"><span class="pre">HPOTerm</span></code> contains various metadata about the term, as well as pointers to its parents and children terms. You can access its information-content, calculate similarity scores to other terms, find the shortest or longes connection between two terms. List all associated genes or diseases, etc.</p>
<section id="id1">
<h5>Examples:<a class="headerlink" href="#id1" title="Permalink to this heading"></a></h5>
<p>Basic functionalities of an HPO-Term</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pyhpo</span> <span class="kn">import</span> <span class="n">Ontology</span>

<span class="c1"># initilize the Ontology ()</span>
<span class="n">_</span> <span class="o">=</span> <span class="n">Ontology</span><span class="p">()</span>

<span class="c1"># Retrieve a term e.g. via its HPO-ID</span>
<span class="n">term</span> <span class="o">=</span> <span class="n">Ontology</span><span class="o">.</span><span class="n">get_hpo_object</span><span class="p">(</span><span class="s1">&#39;Scoliosis&#39;</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="n">term</span><span class="p">)</span>
<span class="c1">#&gt; HP:0002650 | Scoliosis</span>

<span class="c1"># Get information content from Term &lt;--&gt; Omim associations</span>
<span class="n">term</span><span class="o">.</span><span class="n">information_content</span><span class="p">[</span><span class="s1">&#39;omim&#39;</span><span class="p">]</span>
<span class="c1">#&gt; 2.39</span>

<span class="c1"># Show how many genes are associated to the term</span>
<span class="c1"># (Note that this includes indirect associations, associations</span>
<span class="c1"># from children terms to genes.)</span>
<span class="nb">len</span><span class="p">(</span><span class="n">term</span><span class="o">.</span><span class="n">genes</span><span class="p">)</span>
<span class="c1">#&gt; 947</span>

<span class="c1"># Show how many Omim Diseases are associated to the term</span>
<span class="c1"># (Note that this includes indirect associations, associations</span>
<span class="c1"># from children terms to diseases.)</span>
<span class="nb">len</span><span class="p">(</span><span class="n">term</span><span class="o">.</span><span class="n">omim_diseases</span><span class="p">)</span>
<span class="c1">#&gt; 730</span>

<span class="c1"># Get a list of all parent terms</span>
<span class="k">for</span> <span class="n">p</span> <span class="ow">in</span> <span class="n">term</span><span class="o">.</span><span class="n">parents</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">p</span><span class="p">)</span>
<span class="c1">#&gt; HP:0010674 | Abnormality of the curvature of the vertebral column</span>

<span class="c1"># Get a list of all children terms</span>
<span class="k">for</span> <span class="n">p</span> <span class="ow">in</span> <span class="n">term</span><span class="o">.</span><span class="n">children</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">p</span><span class="p">)</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">HP:0002943 | Thoracic scoliosis</span>
<span class="sd">HP:0008458 | Progressive congenital scoliosis</span>
<span class="sd">HP:0100884 | Compensatory scoliosis</span>
<span class="sd">HP:0002944 | Thoracolumbar scoliosis</span>
<span class="sd">HP:0002751 | Kyphoscoliosis</span>
<span class="sd">&quot;&quot;&quot;</span>
</pre></div>
</div>
<p><em>(This script is complete, it should run “as is”)</em></p>
<p>Some additional functionality, working with more than one term</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pyhpo</span> <span class="kn">import</span> <span class="n">Ontology</span>
<span class="n">_</span> <span class="o">=</span> <span class="n">Ontology</span><span class="p">()</span>
<span class="n">term</span> <span class="o">=</span> <span class="n">Ontology</span><span class="o">.</span><span class="n">get_hpo_object</span><span class="p">(</span><span class="s1">&#39;Scoliosis&#39;</span><span class="p">)</span>

<span class="c1"># Let&#39;s get a second term, this time using it HPO-ID</span>
<span class="n">term_2</span> <span class="o">=</span> <span class="n">Ontology</span><span class="o">.</span><span class="n">get_hpo_object</span><span class="p">(</span><span class="s1">&#39;HP:0009121&#39;</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="n">term_2</span><span class="p">)</span>
<span class="c1">#&gt; HP:0009121 | Abnormal axial skeleton morphology</span>

<span class="c1"># Check if the Scoliosis is a direct or indirect child</span>
<span class="c1"># of Abnormal axial skeleton morphology</span>

<span class="n">term</span><span class="o">.</span><span class="n">child_of</span><span class="p">(</span><span class="n">term_2</span><span class="p">)</span>
<span class="c1">#&gt; True</span>

<span class="c1"># or vice versa</span>
<span class="n">term_2</span><span class="o">.</span><span class="n">parent_of</span><span class="p">(</span><span class="n">term</span><span class="p">)</span>
<span class="c1">#&gt; True</span>

<span class="c1"># show all nodes between two term:</span>
<span class="n">path</span> <span class="o">=</span> <span class="n">term</span><span class="o">.</span><span class="n">path_to_other</span><span class="p">(</span><span class="n">term_2</span><span class="p">)</span>
<span class="k">for</span> <span class="n">t</span> <span class="ow">in</span> <span class="n">path</span><span class="p">[</span><span class="mi">1</span><span class="p">]:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">t</span><span class="p">)</span>

<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">HP:0002650 | Scoliosis</span>
<span class="sd">HP:0010674 | Abnormality of the curvature of the vertebral column</span>
<span class="sd">HP:0000925 | Abnormality of the vertebral column</span>
<span class="sd">HP:0009121 | Abnormal axial skeleton morphology</span>
<span class="sd">&quot;&quot;&quot;</span>

<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Steps from Term 1 to Term 2: </span><span class="si">{</span><span class="n">path</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="c1">#&gt; Steps from Term 1 to Term 2: 3</span>


<span class="c1"># Calculate the similarity between two terms</span>
<span class="n">term</span><span class="o">.</span><span class="n">similarity_score</span><span class="p">(</span><span class="n">term_2</span><span class="p">)</span>
<span class="c1">#&gt; 0.442</span>
</pre></div>
</div>
<p><em>(This script is complete, it should run “as is”)</em></p>
</section>
</section>
<section id="ontology">
<h4>Ontology<a class="headerlink" href="#ontology" title="Permalink to this heading"></a></h4>
<p>The <code class="docutils literal notranslate"><span class="pre">Ontology</span></code> contains all HPO terms, their connections to each other and associations to genes and diseases.
It provides some helper functions for <code class="docutils literal notranslate"><span class="pre">HPOTerm</span></code> search functionality</p>
<section id="id2">
<h5>Examples<a class="headerlink" href="#id2" title="Permalink to this heading"></a></h5>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pyhpo</span> <span class="kn">import</span> <span class="n">Ontology</span><span class="p">,</span> <span class="n">HPOSet</span>

<span class="c1"># initilize the Ontology (this must be done only once)</span>
<span class="n">_</span> <span class="o">=</span> <span class="n">Ontology</span><span class="p">()</span>

<span class="c1"># Get a term based on its name</span>
<span class="n">term</span> <span class="o">=</span> <span class="n">Ontology</span><span class="o">.</span><span class="n">get_hpo_object</span><span class="p">(</span><span class="s1">&#39;Scoliosis&#39;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">term</span><span class="p">)</span>
<span class="c1">#&gt; HP:0002650 | Scoliosis</span>

<span class="c1"># ...or based on HPO-ID</span>
<span class="n">term</span> <span class="o">=</span> <span class="n">Ontology</span><span class="o">.</span><span class="n">get_hpo_object</span><span class="p">(</span><span class="s1">&#39;HP:0002650&#39;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">term</span><span class="p">)</span>
<span class="c1">#&gt; HP:0002650 | Scoliosis</span>

<span class="c1"># ...or based on its index</span>
<span class="n">term</span> <span class="o">=</span> <span class="n">Ontology</span><span class="o">.</span><span class="n">get_hpo_object</span><span class="p">(</span><span class="mi">2650</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">term</span><span class="p">)</span>
<span class="c1">#&gt; HP:0002650 | Scoliosis</span>

<span class="c1"># shortcut to retrieve a term based on its index</span>
<span class="n">term</span> <span class="o">=</span> <span class="n">Ontology</span><span class="p">[</span><span class="mi">2650</span><span class="p">]</span>
<span class="nb">print</span><span class="p">(</span><span class="n">term</span><span class="p">)</span>
<span class="c1">#&gt; HP:0002650 | Scoliosis</span>

<span class="c1"># Search for term</span>
<span class="k">for</span> <span class="n">term</span> <span class="ow">in</span> <span class="n">Ontology</span><span class="o">.</span><span class="n">search</span><span class="p">(</span><span class="s1">&#39;olios&#39;</span><span class="p">):</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">term</span><span class="p">)</span>

<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">HP:0002211 | White forelock</span>
<span class="sd">HP:0002290 | Poliosis</span>
<span class="sd">HP:0002650 | Scoliosis</span>
<span class="sd">HP:0002751 | Kyphoscoliosis</span>
<span class="sd">HP:0002943 | Thoracic scoliosis</span>
<span class="sd">HP:0002944 | Thoracolumbar scoliosis</span>
<span class="sd">HP:0003423 | Thoracolumbar kyphoscoliosis</span>
<span class="sd">HP:0004619 | Lumbar kyphoscoliosis</span>
<span class="sd">HP:0004626 | Lumbar scoliosis</span>
<span class="sd">HP:0005659 | Thoracic kyphoscoliosis</span>
<span class="sd">HP:0008453 | Congenital kyphoscoliosis</span>
<span class="sd">HP:0008458 | Progressive congenital scoliosis</span>
<span class="sd">HP:0100884 | Compensatory scoliosis</span>
<span class="sd">&quot;&quot;&quot;</span>
</pre></div>
</div>
<p><em>(This script is complete, it should run “as is”)</em></p>
<p>The Ontology is a Singleton and should only be initiated once.
It can be reused across several modules, e.g:</p>
<p><code class="docutils literal notranslate"><span class="pre">main.py</span></code></p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pyhpo</span> <span class="kn">import</span> <span class="n">Ontology</span><span class="p">,</span> <span class="n">HPOSet</span>

<span class="kn">import</span> <span class="nn">module2</span>

<span class="c1"># initilize the Ontology</span>
<span class="n">_</span> <span class="o">=</span> <span class="n">Ontology</span><span class="p">()</span>

<span class="k">if</span> <span class="vm">__name__</span> <span class="o">==</span> <span class="s1">&#39;__main__&#39;</span><span class="p">:</span>
    <span class="n">module2</span><span class="o">.</span><span class="n">find_term</span><span class="p">(</span><span class="s1">&#39;Compensatory scoliosis&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">module2.py</span></code></p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pyhpo</span> <span class="kn">import</span> <span class="n">Ontology</span>

<span class="k">def</span> <span class="nf">find_term</span><span class="p">(</span><span class="n">term</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">Ontology</span><span class="o">.</span><span class="n">get_hpo_object</span><span class="p">(</span><span class="n">term</span><span class="p">)</span>
</pre></div>
</div>
</section>
</section>
<section id="hposet">
<h4>HPOSet<a class="headerlink" href="#hposet" title="Permalink to this heading"></a></h4>
<p>An <code class="docutils literal notranslate"><span class="pre">HPOSet</span></code> is a collection of <code class="docutils literal notranslate"><span class="pre">HPOTerm</span></code> and can be used to represent e.g. a patient’s clinical information. It provides APIs for filtering, comparisons to other <code class="docutils literal notranslate"><span class="pre">HPOSet</span></code> and term/gene/disease enrichments.</p>
<section id="id3">
<h5>Examples:<a class="headerlink" href="#id3" title="Permalink to this heading"></a></h5>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pyhpo</span> <span class="kn">import</span> <span class="n">Ontology</span><span class="p">,</span> <span class="n">HPOSet</span>

<span class="c1"># initilize the Ontology</span>
<span class="n">_</span> <span class="o">=</span> <span class="n">Ontology</span><span class="p">()</span>

<span class="c1"># create HPOSets, corresponding to</span>
<span class="c1"># e.g. the clinical information of a patient</span>
<span class="c1"># You can initiate an HPOSet using either</span>
<span class="c1"># - HPO-ID: &#39;HP:0002943&#39;</span>
<span class="c1"># - HPO-Name: &#39;Scoliosis&#39;</span>
<span class="c1"># - HPO-ID (int): 2943</span>

<span class="n">ci_1</span> <span class="o">=</span> <span class="n">HPOSet</span><span class="o">.</span><span class="n">from_queries</span><span class="p">([</span>
    <span class="s1">&#39;HP:0002943&#39;</span><span class="p">,</span>
    <span class="s1">&#39;HP:0008458&#39;</span><span class="p">,</span>
    <span class="s1">&#39;HP:0100884&#39;</span><span class="p">,</span>
    <span class="s1">&#39;HP:0002944&#39;</span><span class="p">,</span>
    <span class="s1">&#39;HP:0002751&#39;</span>
<span class="p">])</span>

<span class="n">ci_2</span> <span class="o">=</span> <span class="n">HPOSet</span><span class="o">.</span><span class="n">from_queries</span><span class="p">([</span>
    <span class="s1">&#39;HP:0002650&#39;</span><span class="p">,</span>
    <span class="s1">&#39;HP:0010674&#39;</span><span class="p">,</span>
    <span class="s1">&#39;HP:0000925&#39;</span><span class="p">,</span>
    <span class="s1">&#39;HP:0009121&#39;</span>
<span class="p">])</span>

<span class="c1"># Compare the similarity</span>
<span class="n">ci_1</span><span class="o">.</span><span class="n">similarity</span><span class="p">(</span><span class="n">ci_2</span><span class="p">)</span>
<span class="c1">#&gt; 0.7593552670152157</span>

<span class="c1"># Remove all non-leave nodes from a set</span>
<span class="n">ci_leaf</span> <span class="o">=</span> <span class="n">ci_2</span><span class="o">.</span><span class="n">child_nodes</span><span class="p">()</span>
<span class="nb">len</span><span class="p">(</span><span class="n">ci_2</span><span class="p">)</span>
<span class="c1">#&gt; 4</span>
<span class="nb">len</span><span class="p">(</span><span class="n">ci_leaf</span><span class="p">)</span>
<span class="c1">#&gt; 1</span>
<span class="n">ci_2</span>
<span class="c1">#&gt; HPOSet.from_serialized(&quot;925+2650+9121+10674&quot;)</span>
<span class="n">ci_leaf</span>
<span class="c1">#&gt; HPOSet.from_serialized(&quot;2650&quot;)</span>

<span class="c1"># Check the information content of an HPOSet</span>
<span class="n">ci_1</span><span class="o">.</span><span class="n">information_content</span><span class="p">()</span>
<span class="sd">&quot;&quot;&quot;</span>
<span class="sd">{</span>
<span class="sd">    &#39;mean&#39;: 6.571224974009769,</span>
<span class="sd">    &#39;total&#39;: 32.856124870048845,</span>
<span class="sd">    &#39;max&#39;: 8.97979449089521,</span>
<span class="sd">    &#39;all&#39;: [5.98406221734122, 8.286647310335265, 8.97979449089521, 5.5458072864100645, 4.059813565067086]</span>
<span class="sd">}</span>
<span class="sd">&quot;&quot;&quot;</span>
</pre></div>
</div>
<p><em>(This script is complete, it should run “as is”)</em></p>
</section>
</section>
<section id="get-genes-enriched-in-an-hposet">
<h4>Get genes enriched in an <code class="docutils literal notranslate"><span class="pre">HPOSet</span></code><a class="headerlink" href="#get-genes-enriched-in-an-hposet" title="Permalink to this heading"></a></h4>
<section id="id4">
<h5>Examples:<a class="headerlink" href="#id4" title="Permalink to this heading"></a></h5>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pyhpo</span> <span class="kn">import</span> <span class="n">Ontology</span><span class="p">,</span> <span class="n">HPOSet</span>
<span class="kn">from</span> <span class="nn">pyhpo.stats</span> <span class="kn">import</span> <span class="n">EnrichmentModel</span>

<span class="c1"># initilize the Ontology</span>
<span class="n">_</span> <span class="o">=</span> <span class="n">Ontology</span><span class="p">()</span>

<span class="n">ci</span> <span class="o">=</span> <span class="n">HPOSet</span><span class="o">.</span><span class="n">from_queries</span><span class="p">([</span>
    <span class="s1">&#39;HP:0002943&#39;</span><span class="p">,</span>
    <span class="s1">&#39;HP:0008458&#39;</span><span class="p">,</span>
    <span class="s1">&#39;HP:0100884&#39;</span><span class="p">,</span>
    <span class="s1">&#39;HP:0002944&#39;</span><span class="p">,</span>
    <span class="s1">&#39;HP:0002751&#39;</span>
<span class="p">])</span>

<span class="n">gene_model</span> <span class="o">=</span> <span class="n">EnrichmentModel</span><span class="p">(</span><span class="s1">&#39;gene&#39;</span><span class="p">)</span>
<span class="n">genes</span> <span class="o">=</span> <span class="n">gene_model</span><span class="o">.</span><span class="n">enrichment</span><span class="p">(</span><span class="n">method</span><span class="o">=</span><span class="s1">&#39;hypergeom&#39;</span><span class="p">,</span> <span class="n">hposet</span><span class="o">=</span><span class="n">ci</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="n">genes</span><span class="p">[</span><span class="mi">0</span><span class="p">][</span><span class="s1">&#39;item&#39;</span><span class="p">])</span>
<span class="c1">#&gt; PAPSS2</span>
</pre></div>
</div>
<p><em>(This script is complete, it should run “as is”)</em></p>
<p>For a more detailed description of how to use PyHPO, visit the <a class="reference external" href="https://centogene.github.io/pyhpo/">PyHPO Documentation</a>.</p>
</section>
</section>
</section>
<section id="contributing">
<h3>Contributing<a class="headerlink" href="#contributing" title="Permalink to this heading"></a></h3>
<p>Yes, please do so. We appreciate any help, suggestions for improvement or other feedback. Just create a pull-request or open an issue.</p>
</section>
<section id="license">
<h3>License<a class="headerlink" href="#license" title="Permalink to this heading"></a></h3>
<p>PyHPO is released under the <a class="reference external" href="http://www.opensource.org/licenses/mit-license.php">MIT license</a>.</p>
<p>PyHPO is using the Human Phenotype Ontology. Find out more at <a class="reference external" href="http://www.human-phenotype-ontology.org">http://www.human-phenotype-ontology.org</a></p>
<p>Sebastian Köhler, Leigh Carmody, Nicole Vasilevsky, Julius O B Jacobsen, et al. Expansion of the Human Phenotype Ontology (HPO) knowledge base and resources. Nucleic Acids Research. (2018) doi: 10.1093/nar/gky1105</p>
</section>
</section>
</section>
<section id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this heading"></a></h1>
<ul class="simple">
<li><p><a class="reference internal" href="genindex.html"><span class="std std-ref">Index</span></a></p></li>
<li><p><a class="reference internal" href="py-modindex.html"><span class="std std-ref">Module Index</span></a></p></li>
<li><p><a class="reference internal" href="search.html"><span class="std std-ref">Search Page</span></a></p></li>
</ul>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="hpoterm.html" class="btn btn-neutral float-right" title="HPOTerm" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2021, CENTOGENE GmbH.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>